mix.exs

```diff
@@ -23,13 +24,60 @@ defmodule Floki.Mixfile do
   end
 
   defp deps do
+    # Needed to avoid installing unnecessary deps on travis
+    parsers =
+      case System.get_env("PARSER") do
+        nil -> [:fast_html, :html5ever]
+        parser -> [String.to_atom(parser)]
+      end
+      |> Enum.map(fn name -> {name, ">= 0.0.0", optional: true, only: [:dev, :test]} end)
+
     [
       {:html_entities, "~> 0.5.0"},
       {:earmark, "~> 1.2", only: :dev},
       {:ex_doc, "~> 0.18", only: :dev},
       {:credo, ">= 0.0.0", only: [:dev, :test]},
       {:inch_ex, ">= 0.0.0", only: :docs}
-    ]
+    ] ++ parsers
+  end
+
+  defp aliases do
+    parsers = get_parsers()
+
+    {aliases, cli_names} =
+      Enum.map_reduce(parsers, [], fn parser, acc ->
+        cli_name =
+          parser
+          |> Module.split()
+          |> List.last()
+          |> Macro.underscore()
+
+        {{:"test.#{cli_name}", &test_with_parser(parser, &1)}, [cli_name | acc]}
+      end)
+
+    aliases
+    |> Keyword.put(:test, &test_with_parser(cli_names, &1))
+  end
+
+  # Hardcoded because we can't load the floki application and get the module list at this point.
+  defp get_parsers() do
+    [Floki.HTMLParser.Mochiweb, Floki.HTMLParser.FastHtml, Floki.HTMLParser.Html5ever]
+  end
+
+  # Hack: If Mix.Task.Test.run is called the second time, it won't run any tests
```

Comment:
This is probably related to the `ExUnit` that is started at `test/test_helper.exs` 🤔 
It's OK to leave without the line comment here.